src/monster.cpp

```diff
@@ -1277,7 +1281,7 @@ void monster::melee_attack( Creature &target, float accuracy )
                 //~ $3s is target armor name.
                 add_msg( _( "The %1$s hits %2$s but is stopped by its %3$s." ),
                          name().c_str(),
-                         target.disp_name( true ).c_str(),
+                         target.disp_name().c_str(),
```

Comment:
You may remove all that `.c_str()` while you are here.

Reply:
Yeah, I know, there are just too many of them in the function, so I doubt if I should remove them all in this PR in question.